pa-rserve/src/main/java/org/ow2/parserve/util/rsession/StartRserve.java

```diff
@@ -0,0 +1,253 @@
+        logger.info("Waiting for Rserve to start ...");
+        boolean startRserve;
+        if (daemon) {
+            startRserve = doInR("library(Rserve);" + (preloadFile != null ? "source('" + Utils.toRpath(preloadFile) + "');" : "") + "Rserve(" + (debug ? "TRUE" : "FALSE") + ",args='" + rsrvargs + "')", cmd, rargs, false, null, null);
+        } else {
+            startRserve = doInR("library(Rserve);" + (preloadFile != null ? "source('" + Utils.toRpath(preloadFile) + "');" : "") + "run.Rserve(" + rsrvargs + ")", cmd, rargs, false, null, null);
+        }
+        if (startRserve) {
+            logger.info("Rserve startup done, let us try to connect ...");
+        } else {
+            return false;
+        }
+
+        // try up to 120 seconds before giving up. We can be conservative here, because at this point the process execution
+        // itself was successful and the startup is usually asynchronous
+        int attempts = (timeout > 0 ? timeout : 120);
+        while (attempts > 0) {
+            try {
+                RConnection c = null;
+                int port = -1;
+                if (rsrvargs.contains("--RS-port")) {
+                    String rsport = rsrvargs.split("--RS-port")[1].trim().split(" ")[0];
+                    port = Integer.parseInt(rsport);
+                } else if (rsrvargs.contains("port")) {
+                    String rsport = rsrvargs.split("port")[1].trim().split("=")[1].trim().split(",")[0].trim();
+                    port = Integer.parseInt(rsport);
+                }
+                logger.info("Trying to connect to RServe on port : " + port);
+                if (rsrvargs.contains("--RS-port") || rsrvargs.contains("port")) {
+                    c = new RConnection("localhost", port);
+                } else {
+                    c = new RConnection("localhost");
+                }
+                logger.info("Rserve is running.");
+                c.close();
+                return true;
+            } catch (Exception e2) {
+                logger.debug("Try failed with: " + e2.getMessage());
+            }
+            /* a safety sleep just in case the start up is delayed or asynchronous */
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException ix) {
+            }
+
+            attempts--;
+        }
+        return false;
+    }
+
+    /**
+     * checks whether Rserve is running and if that's not the case it attempts to start it using the defaults for the platform where it is run on.
+     * This method is meant to be set-and-forget and cover most default setups. For special setups you may get more control over R with <code>launchRserve</code> instead.
+     */
+    public static boolean checkLocalRserve() {
+        if (isRserveRunning()) {
+            return true;
+        }
+        String osname = System.getProperty("os.name");
+        if (osname != null && osname.length() >= 7 && osname.substring(0, 7).equals("Windows")) {
```

Comment:
Is it not possible to reuse one of our utility class for that? or Apache common?